synapse/handlers/initial_sync.py

```diff
@@ -116,14 +116,13 @@ async def _snapshot_all_rooms(
         now_token = self.hs.get_event_sources().get_current_token()
 
         presence_stream = self.hs.get_event_sources().sources["presence"]
-        pagination_config = PaginationConfig(from_token=now_token)
-        presence, _ = await presence_stream.get_pagination_rows(
-            user, pagination_config.get_source_config("presence"), None
+        presence, _ = await presence_stream.get_new_events(
+            user, from_key=None, include_offline=False
         )
 
-        receipt_stream = self.hs.get_event_sources().sources["receipt"]
-        receipt, _ = await receipt_stream.get_pagination_rows(
-            user, pagination_config.get_source_config("receipt"), None
+        joined_rooms = [r.room_id for r in room_list if r.membership == Membership.JOIN]
+        receipt = await self.store.get_linearized_receipts_for_rooms(
+            joined_rooms, to_key=int(now_token.receipt_key),
         )
```

Comment:
I'm having a bit of trouble following how the keys were modified as part of this code, previously we have:

```python
pagination_config = PaginationConfig(from_token=now_token)
config = pagination_config.get_source_config("receipt")
to_key = int(config.from_key)

if config.to_key:
	from_key = int(config.to_key)
else:
	from_key = None
```

Previously:

* `get_source_config` was getting `now_token.receipt_key` (and saving it as `from_key` since that's how `pagination_config` was instantiated).
* Then `to_key` was set as `int(config.from_key)`.
* Since `config.to_key` was never set (since `pagination_config.to_key` is unset) then `from_key` should be `None`.

So this looks equivalent.